merkledag/README.md

```diff
@@ -0,0 +1,144 @@
+
+The logical representation is serialized into raw bytes using _protocol buffers_, a serialization format.
+
+## Discussion
+
+### Real World Examples
+
+Many successful distributed systems employ specialized merkledags at their core:
+- merkle trees -- a special case of the merkle dag -- are a well known cryptographic technique used to authenticate large amounts of data. The original use case involved one-time lamport signatures.
+- SFS-RO turns a unix filesystem into a merkledag, constructing a secure, distributed filesystem.
+- git uses a merkledag to enable distributed version control and source code management. Other DVCSes, such as mercurial and monotone, also feature a merkledag.
+- plan9 uses a merkledag to construct its snapshotting filesystems -- Fossil and Venti.
+- bittorrent uses a merkledag to provide secure and short infohash links to its downloadable torrents.
+- Google Wave -- a distributed communications platform -- used a merkledag to construct its commutative operational transforms, and enable convergent distributed collaboration. This functionality has since been folded into Google Docs.
+- bitcoin uses a merkledag to construct the blockchain, a shared append-only ledger with convergent distributed consensus.
+- Tahoe-LAFS uses a merkledag to construct a secure, distributed, capability filesystem based on the least-authority principle.
+
+(NOTE: please suggest other systems to reference here.)
+
+### Thin Waist of Data Structures
+
+At its core, IPFS provides the merkledag as a primitive (or "internet layer") to build sophisticated applications easily. It is a "thin-waist" for secure, distributed applications, which -- by agreeing to follow the common format -- can then run across any replication, routing, and transport protocols. To draw an analogy, this is like the "thin-waist" IP provided to connect hosts across medium-specific networks.
+
+![](mdag.waist.png)
+
+![](ip.waist.png)
+
+This kind of modularity enables complicated and powerful applications to be built with little effort on top of a common base. All the complexity of authentication, distribution, replication, routing, and transport can be pulled in from other protocols and tools. This type of modularity is what made the layered internet -- the TCP/IP stack -- so tremendously powerful.
+
+### Web of Data Structures
+
+In a sense, IPFS is a "web of data-structures", with the merkledag as the common denominator. Agreeing upon a format allows linking different authenticated datastructures to each other, enabling sophisticated distributed applications to easily construct, distribute, and link their data.
+
+### Linked Data
+
+The merkledag is a type of Linked-Data. The links do not follow the standard URI format, and instead opt for a more general and flexible UNIX filesystem path format, but the power is all there. One can trivially map formats like JSON-LD directly onto IPFS (IPFS-LD), making IPFS applications capable of using the full-power of the semantic web.
```

Comment:
this part doesnt say anything about "Typing via entries in Links". this part talks about JSON-LD, which the base merkledag format supports out of the box, because JSON-LD just adds one `@context` entry mapped to a spec written in JSON. an equivalent `IPFS-LD` would work similarly.